src/validation.cpp

```diff
@@ -3228,9 +3234,20 @@ bool CChainState::AcceptBlockHeader(const CBlockHeader& block, CValidationState&
             return true;
         }
 
-        if (!CheckBlockHeader(block, state))
+        if (!CheckBlockHeader(block, state, aggPubkeys))
             return error("%s: Consensus::CheckBlockHeader: %s, %s", __func__, hash.ToString(), FormatStateMessage(state));
 
+        //if this header was valid and has an aggpubkey change remember it until we finish processing the headers message
+        if(aggPubkeys && (TAPYRUS_XFIELDTYPES)block.xfieldType == TAPYRUS_XFIELDTYPES::AGGPUBKEY
+          && block.xfield.size() == CPubKey::COMPRESSED_PUBLIC_KEY_SIZE
+          && CPubKey(block.xfield.begin(), block.xfield.end()) != aggPubkeys->rbegin()->aggpubkey)
+        {
+            aggPubkeyAndHeight x;
```

Comment:
Although not directly related to this change, it would be better to rename this struct to `AggPubkeyAndHeight`.